### PR TITLE
Update HDF to 1_13_0 release

### PR DESCRIPTION
**Pull request by hjmjohnson**
## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by dzenanz** [comment]
Does this supersede #2832?

**Comment by seanm** [comment]
HDF5 1.13 is an ***experimental release***, I don't think suitable for merging into ITK yet.  See: https://www.hdfgroup.org/2021/12/hdf5-1-13-0-introducing-experimental-releases/

**Comment by seanm** [comment]
#2832 is so very close to being ready. It just needs someone that knows cmake, to get it working as shared libs.  As static libs, it builds with every build setting permutation I could find.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h [hunk 1]
```diff
@@ -4047,20 +3742,1430 @@ e.g. remove mangling for H5Fget_info but leave it for H5Fget_info1 and H5Fget_in
...
-#define HDfprintf itk_HDfprintf
 #define HDrand itk_HDrand
 #define HDsrand itk_HDsrand
-#define HDstrtoll itk_HDstrtoll
 #define Nflock itk_Nflock
 #define Pflock itk_Pflock
-#define epoch_marker_class itk_epoch_marker_class
+#define VERS_RELEASE_EXCEPTIONS itk_VERS_RELEASE_EXCEPTIONS
+#define VERS_RELEASE_EXCEPTIONS_SIZE itk_VERS_RELEASE_EXCEPTIONS_SIZE
+#define _ZN2H510AbstractDsC2Ev itk__ZN2H510AbstractDsC2Ev
```
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h]: These look like compiler mangled symbols, I am not sure they should be listed/mangled here.

**Comment by hjmjohnson** [comment]
Abandoning. HDF5 is such a pain!  Hopefuly we can get #2832 to work.

**Pull request closed without merging**